src/lib/openjp2/pi.c

```diff
@@ -1237,7 +1237,13 @@ opj_pi_iterator_t *opj_pi_create_decode(opj_image_t *p_image,
 	l_current_pi = l_pi;
 
 	/* memory allocation for include */
-	l_current_pi->include = (OPJ_INT16*) opj_calloc((l_tcp->numlayers +1) * l_step_l, sizeof(OPJ_INT16));
+	/* prevent an integer overflow issue */
+	l_current_pi->include = 00;
+	if (l_step_l && l_tcp->numlayers < ((SIZE_MAX)-1) / sizeof(OPJ_INT16) / l_step_l - 1)
```

Comment:
Sorry for my late comment.
The check with `sizeof(OPJ_INT16)` is unnecessary since it will be done by `opj_calloc` (through a call to `calloc`)
We can get rid of the check of `l_step_l` against `0` since `l_tcp->numlayers + 1U > 0U` and `l_tcp->numlayers + 1U` can't overflow.

The check can thus become:
`if (l_step_l <= ((SIZE_MAX) / (l_tcp->numlayers + 1U)))`